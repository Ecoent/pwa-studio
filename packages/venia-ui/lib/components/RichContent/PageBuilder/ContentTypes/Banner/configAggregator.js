import {getMargin, getBackgroundImages, getBorder, getPadding} from '../../utils';

export default (node, props) => {
    console.log(props.appearance, node);
    const wrapper = node.querySelector('[data-element="wrapper"]');
    const overlay = node.querySelector('[data-element="overlay"]');
    const response = {
        minHeight: overlay.style.minHeight ? overlay.style.minHeight : null,
        content: node.querySelector('[data-element="content"]').innerHTML,
        showButton: node.getAttribute('data-show-button'),
        showOverlay: node.getAttribute('data-show-overlay'),
        ...getBackgroundImages(wrapper),
        ...getBorder(wrapper),
        ...getMargin(node),
        ...getPadding(overlay),
    };
    console.log(response);
    return response;
};
