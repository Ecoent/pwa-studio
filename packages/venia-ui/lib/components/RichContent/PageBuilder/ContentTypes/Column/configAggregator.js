import {
    getAdvanced,
    getBackgroundImages,
    getBorder,
    getCssClasses,
    getMargin,
    getTextAlign,
    getVerticalAlignment
} from '../../utils';

export default node => {
    return {
        minHeight: node.style.minHeight ? node.style.minHeight : null,
        display: node.style.display,
        width: node.style.width,
        backgroundColor: node.style.backgroundColor,
        ...getAdvanced(node),
        ...getBackgroundImages(node),
        ...getBorder(node),
        ...getCssClasses(node),
        ...getMargin(node),
        ...getTextAlign(node),
        ...getVerticalAlignment(node)
    };
};
