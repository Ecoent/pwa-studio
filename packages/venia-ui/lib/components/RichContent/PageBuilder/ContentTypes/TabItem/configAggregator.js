import {
    getAdvanced,
    getBackgroundImages,
    getVerticalAlignment
} from '../../utils';

export default node => {
    return {
        tabName: node.getAttribute('data-tab-name'),
        minHeight: node.style.minHeight,
        ...getVerticalAlignment(node),
        ...getBackgroundImages(node),
        ...getAdvanced(node)
    };
};
