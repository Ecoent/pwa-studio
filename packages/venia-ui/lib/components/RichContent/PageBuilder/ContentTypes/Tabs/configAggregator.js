import {
    getAdvanced,
    getBackgroundImages,
    getVerticalAlignment
} from '../../utils';

export default node => {
    const navigationEl = node.childNodes[0];
    const headerEls = navigationEl.childNodes;
    const headers = Array.from(headerEls, headerEl => headerEl.textContent);

    const contentEl = node.childNodes[1];

    return {
        defaultIndex: node.getAttribute('data-active-tab')
            ? parseInt(node.getAttribute('data-active-tab'), 10)
            : 0,
        headers,
        minHeight: contentEl.style.minHeight,
        ...getVerticalAlignment(node),
        backgroundColor: node.style.backgroundColor,
        ...getBackgroundImages(node),
        ...getAdvanced(node)
    };
};
