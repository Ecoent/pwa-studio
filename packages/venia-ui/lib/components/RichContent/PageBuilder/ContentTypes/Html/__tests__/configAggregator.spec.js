import configAggregator from '../configAggregator';

test('config is aggregated correctly for html content type', () => {
    const node = document.createElement('div');
    node.innerHTML = `<div data-content-type="row" data-appearance="contained" data-element="main"><div data-enable-parallax="0" data-parallax-speed="0.5" data-background-images="{}" data-element="inner" style="justify-content: flex-start; display: flex; flex-direction: column; background-position: left top; background-size: cover; background-repeat: no-repeat; background-attachment: scroll; border-style: none; border-width: 1px; border-radius: 0px; margin: 0px 0px 10px; padding: 10px;"><div data-content-type="html" data-appearance="default" data-element="main" style="border-style: none; border-width: 1px; border-radius: 0px; margin: 0px; padding: 0px;">&lt;button href='{{config path="web/unsecure/base_url"}}'&gt;{{widget type="Magento\\Cms\\Block\\Widget\\Page\\Link" anchor_text="Anchor Custom Text" title="Anchor Custom Title" template="widget/link/link_inline.phtml" page_id="2"}}&lt;/button&gt;&lt;img src="{{media url=catalog/category/carefree.jpg}}" alt="" /&gt;</div></div></div>`;
    const config = configAggregator(node.childNodes[0]);

    expect(config).toMatchSnapshot();
});
