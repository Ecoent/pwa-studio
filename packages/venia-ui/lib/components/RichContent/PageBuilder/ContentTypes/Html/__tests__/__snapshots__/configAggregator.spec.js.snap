// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config is aggregated correctly for html content type 1`] = `
Object {
  "border": "",
  "borderColor": "",
  "borderRadius": "",
  "borderWidth": "",
  "cssClasses": Array [],
  "html": "undefined",
  "marginBottom": "",
  "marginLeft": "",
  "marginRight": "",
  "marginTop": "",
  "paddingBottom": "",
  "paddingLeft": "",
  "paddingRight": "",
  "paddingTop": "",
  "textAlign": "",
}
`;
